pkg/deploy/controller/deployment/controller.go

```diff
@@ -141,14 +136,34 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 			glog.V(4).Infof("Cancelling %d deployer pods for deployment %s", len(deployerPods), deployutil.LabelForDeployment(deployment))
 			zeroDelay := int64(1)
 			for _, deployerPod := range deployerPods {
-				// Set the ActiveDeadlineSeconds on the pod so it's terminated very soon.
-				if deployerPod.Spec.ActiveDeadlineSeconds == nil || *deployerPod.Spec.ActiveDeadlineSeconds != zeroDelay {
-					deployerPod.Spec.ActiveDeadlineSeconds = &zeroDelay
-					if _, err := c.podClient.updatePod(deployerPod.Namespace, &deployerPod); err != nil {
-						c.recorder.Eventf(deployment, "failedCancellation", "Error cancelling deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
-						return fmt.Errorf("couldn't cancel deployer pod %s for deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+				if deployerPod.Status.Phase == kapi.PodPending {
+					// If the pod hasn't yet been scheduled, just delete it.
+					err := c.podClient.deletePod(deployerPod.Namespace, deployerPod.Name)
+					if err != nil {
+						c.recorder.Eventf(deployment, "failedCancellation", "Error deleting deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+						return fmt.Errorf("couldn't delete deployer pod %s while cancelling deployment %s: %v", deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
+					}
+					// If we deleted the unscheduled primary deployer pod, just
+					// transition to failed immediately, since the DeployerPodController
+					// has no cleanup to do as the deployment never executed.
+					if deployerPod.Name == deployutil.DeployerPodNameForDeployment(deployment.Name) {
```

Comment:
You should mark it failed first, if you get a conflict you'll know it got started.  Then you can delete the pod safely.